packages/common/src/utils/SafeLocalStorage.ts

```diff
@@ -24,7 +32,7 @@ export const SafeLocalStorage = {
     value: SafeLocalStorageItems[Key]
   ): void {
     if (isSafe()) {
-      localStorage.setItem(key, JSON.stringify(value))
+      localStorage.setItem(key, value)
```

Comment:
We shouldn't do the parsing here. An example;

We have `@appkit/active_caip_network` key, which is expected to store `CaipNetwork` type. 

If we set `'@appkit/active_caip_network': string` on `SafeLocalStorageItems`, the `getItem`'s `catch` will return type error bc, if somehow the parsing fails, it'll return `string` which is not what that key stores.

So the `stringify`ing and `parse`ing should be abstracted from this `SafeLocalStorage` utility.